ocs_ci/ocs/utils.py

```diff
@@ -925,45 +956,110 @@ def run_must_gather(log_dir_path, image, command=None, cluster_config=None):
             cluster_config=cluster_config,
         )
     except CommandFailed as ex:
-        log.error(get_helper_pods_output())
         log.error(
             f"Failed during must gather logs! Error: {ex}"
             f"Must-Gather Output: {mg_output}"
         )
+        export_mg_pods_logs(log_dir_path=log_dir_path)
+
     except TimeoutExpired as ex:
-        log.error(get_helper_pods_output())
         log.error(
-            f"Timeout {must_gather_timeout}s for must-gather reached, command"
-            f" exited with error: {ex}"
+            f"Failed during must gather logs! Error: {ex}"
             f"Must-Gather Output: {mg_output}"
         )
+        export_mg_pods_logs(log_dir_path=log_dir_path)
     return mg_output
 
 
-def get_helper_pods_output():
+def export_mg_pods_logs(log_dir_path):
     """
-    Get the output of "oc describe mg-helper pods"
+    Export must gather pods logs
+
+    Args:
+        log_dir_path (str): the path of copying the logs
 
     Returns:
-        str: the output of "oc describe pods mg-helper" and "oc logs mg-helper"
+
+    """
+    get_logs_ocp_mg_pods(log_dir_path)
+    get_helper_pods_output(log_dir_path)
+
+
+def get_logs_ocp_mg_pods(log_dir_path):
+    """
+    Get logs from OCP Must Gather pods
+
+    Args:
+        log_dir_path (str): the path of copying the logs
+
+    """
+    from ocs_ci.ocs.resources.pod import get_all_pods, get_pod_logs
+
+    namespaces = get_namespce_name_by_pattern(pattern="openshift-must-gather")
+    try:
+        for namespace in namespaces:
+            pods_mg_ns = get_all_pods(namespace=namespace)
+            for pod_mg_ns in pods_mg_ns:
+                log.info(
+                    f"*** ocp_mg_pod_name: {pod_mg_ns.name} ocp_mg_pod_namespace: {namespace} ***"
+                )
+
+                file_path_describe = os.path.join(
+                    log_dir_path, f"describe_ocp_mg_{pod_mg_ns.name}.log"
+                )
+                with open(file_path_describe, "w") as df:
+                    df.write(f"ocp mg pod describe:\n{pod_mg_ns.describe()}")
+                log.debug(f"ocp mg pod describe:\n{pod_mg_ns.describe()}")
```

Comment:
instead of calling pod_mg_ns.describe() twice, can we store output to variable and use the same in debug log.

Reply:
done